components/assetstore-controller-manager/pkg/controller/asset/webhook/validation_webhook.go

```diff
@@ -0,0 +1,110 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"github.com/kyma-project/kyma/components/assetstore-controller-manager/pkg/apis/assetstore/v1alpha1"
+	"github.com/kyma-project/kyma/components/assetstore-controller-manager/pkg/assethook"
+	webhookv1alpha1 "github.com/kyma-project/kyma/components/assetstore-controller-manager/pkg/assethook/api/v1alpha1"
+	"io/ioutil"
+	"sync"
+	"time"
+)
+
+//go:generate mockery -name=Validator -output=automock -outpkg=automock -case=underscore
+type Validator interface {
+	Validate(ctx context.Context, basePath string, files []string, asset *v1alpha1.Asset) (ValidationResult, error)
+}
+
+type validationWebhook struct {
+	baseWebhook
+	webhook    assethook.Webhook
+	timeout    time.Duration
+	fileReader func(filename string) ([]byte, error)
+}
+
+type ValidationResult struct {
+	Success  bool
+	Messages []string
+}
+
+func NewValidator(webhook assethook.Webhook, timeout time.Duration) Validator {
+	return &validationWebhook{
+		webhook:    webhook,
+		timeout:    timeout,
+		fileReader: ioutil.ReadFile,
+	}
+}
+
+func (w *validationWebhook) Validate(ctx context.Context, basePath string, files []string, asset *v1alpha1.Asset) (ValidationResult, error) {
+	errCh := make(chan error, 1)
+	rspCh := make(chan webhookv1alpha1.ValidationResponse, 1)
+
+	go func() {
+		defer close(errCh)
+		defer close(rspCh)
+
+		validatorsCount := len(asset.Spec.Source.ValidationWebhookService)
+		var wg sync.WaitGroup
+		wg.Add(validatorsCount)
+
+		for _, validation := range asset.Spec.Source.ValidationWebhookService {
```

Comment:
It is absolutely fine for now, but in future we could introduce workers - it will be safer to limit concurrent connections, if user will have a large number of validators defined. If you think it is the case, please add TODO comment with that.

Reply:
We resigned from concurrency, so it will be simplified.